packages/medusa/src/strategies/batch-jobs/product/export.ts

```diff
@@ -0,0 +1,344 @@
+import { AbstractBatchJobStrategy, IFileService } from "../../../interfaces"
+import { Product, ProductVariant } from "../../../models"
+import { BatchJobService, ProductService } from "../../../services"
+import { BatchJobStatus, CreateBatchJobInput } from "../../../types/batch-job"
+import { defaultAdminProductRelations } from "../../../api/routes/admin/products"
+import { prepareListQuery } from "../../../utils/get-query-config"
+import {
+  ProductExportBatchJob,
+  ProductExportBatchJobContext,
+  ProductExportColumnSchemaDescriptor,
+  productExportSchemaDescriptors,
+} from "./index"
+import { FindProductConfig } from "../../../types/product"
+
+type InjectedDependencies = {
+  manager: EntityManager
+  batchJobService: BatchJobService
+  productService: ProductService
+  fileService: IFileService<never>
+}
+
+export default class ProductExportStrategy extends AbstractBatchJobStrategy<
+  ProductExportStrategy,
+  InjectedDependencies
+> {
+  public static identifier = "product-export-strategy"
+  public static batchType = "product-export"
+
+  protected manager_: EntityManager
+  protected transactionManager_: EntityManager | undefined
+
+  protected readonly batchJobService_: BatchJobService
+  protected readonly productService_: ProductService
+  protected readonly fileService_: IFileService<never>
+
+  protected readonly defaultRelations_ = [...defaultAdminProductRelations]
+  /*
+   *
+   * The dynamic columns corresponding to the lowest level of relations are built later on.
+   * You can have a look at the buildHeader method that take care of appending the other
+   * column descriptors to this map.
+   *
+   */
+  protected readonly columnDescriptors: Map<
+    string,
+    ProductExportColumnSchemaDescriptor
+  > = productExportSchemaDescriptors
+
+  private readonly NEWLINE_ = "\r\n"
+  private readonly DELIMITER_ = ";"
+  private readonly BATCH_SIZE = 50
+
+  constructor({
+    manager,
+    batchJobService,
+    productService,
+    fileService,
+  }: InjectedDependencies) {
+    super({
+      manager,
+      batchJobService,
+      productService,
+      fileService,
+    })
+
+    this.manager_ = manager
+    this.batchJobService_ = batchJobService
+    this.productService_ = productService
+    this.fileService_ = fileService
+  }
+
+  async buildTemplate(): Promise<string> {
+    return ""
+  }
+
+  async preProcessBatchJob(batchJobId: string): Promise<void> {
+    return await this.atomicPhase_(async (transactionManager) => {
+      const batchJob = (await this.batchJobService_
+        .withTransaction(transactionManager)
+        .retrieve(batchJobId)) as ProductExportBatchJob
+
+      const { list_config = {}, filterable_fields = {} } = batchJob.context
+
+      const [productList] = await this.productService_
+        .withTransaction(transactionManager)
+        .listAndCount(filterable_fields, {
+          ...list_config,
+          skip: undefined,
+          take: undefined,
+        } as FindProductConfig)
```

Comment:
thought: could there be a risk of this ending in an oom as we are potentially making a huge request from the db here? Potentially apply a cursor-ish approach?

Reply:
Probably should look by batch size indeed 😅